src/signalfx_metadata.py

```diff
@@ -883,6 +890,39 @@ def plugin_config(conf):
     get_aws_info()
 
 
+def load_persistent_data():
+    """Load persistent data from a specified path"""
+    try:
+        with open(os.path.join(PERSISTENCE_PATH, PERSISTENCE_FILE), 'r') as js:
+            persist = json.load(js)
+            debug("Loaded the following persistent data %s" % persist)
+            if "NEXT_METADATA_SEND" in persist:
+                global NEXT_METADATA_SEND
+                NEXT_METADATA_SEND = persist["NEXT_METADATA_SEND"]
+            if "NEXT_METADATA_SEND_INTERVAL" in persist:
+                global NEXT_METADATA_SEND_INTERVAL
+                NEXT_METADATA_SEND_INTERVAL = \
+                    persist["NEXT_METADATA_SEND_INTERVAL"]
```

Comment:
you need to store and check the hostname against what this running one thinks is the hostname. if it's different you need to restart the metadata send intervals.